[WIP] mobile web: Fix bugs in message action buttons

ab1nash opened the pull request:
The three message actions buttons on the top right of a message box had a
few bugs in mobile web related to visibility as explained in #13642 . This fixes those visibility
issues by adding a semi-opaque background and removing the hover property
for these buttons in mobile web view.

Fixes #13642

Tested on mobile view and normal view.
[TODO] :
Some residual issue with the star's opacity not changing.
The edit button does not appear unless hovered on, beyond which it persists.
* This is not working on the first message of a topic ( possibly because of the topic name).
   Should this be fixed?

[QUERIES]:
Should I keep the background for `message_time`? Looks a little odd otherwise.

ab1nash commented on static/styles/zulip.scss:
```diff
@@ -592,7 +592,7 @@ td.pointer {
     position: absolute;
     right: -105px;
-    line-height: 20px;
+    line-height: 18px;
```
on adding background, message_time was not aligned with the action buttons' background

chrisbobbe replied:
Hmm, with 18px, the time text looks a little high to me; what do you think, @showell?

![image](https://user-images.githubusercontent.com/22248748/75191490-47d8ce80-5707-11ea-8ca6-0e65e04996d3.png)

vs. original, with 20px:

![image](https://user-images.githubusercontent.com/22248748/75191577-72c32280-5707-11ea-86fe-09e9ff24a623.png)

Is this the intended effect?

ab1nash commented:
Mobile view
![Screenshot from 2020-02-03 12-40-00](https://user-images.githubusercontent.com/41019632/73632681-7c1a1b80-4682-11ea-94db-5e3d1bb22e5e.png)
Normal view
![Screenshot from 2020-02-03 12-40-18](https://user-images.githubusercontent.com/41019632/73632682-7c1a1b80-4682-11ea-928b-8982e4ba8d14.png)

timabbott commented:
For the mobile view, I think the intent is to only show the background and buttons when the user has clicked on the message, rather than having them be continuously visible.  Is that what this does?  That's not what the screenshot shows.

I think if we were to make it only appear on click, and change the background color to be lighter, we wouldn't want a background colors on the timestamps.

@chrisbobbe thoughts?

chrisbobbe commented:
I agree with all of that, Tim.

Also, it doesn't seem like quite the right color to give enough contrast for people to read the icons on the left and in the middle; they're in a shade of gray very similar to the background.

Also, we'll need to have this working in night mode, too, could you please include some screenshots for that, as well? Thanks! 🙂

ab1nash commented:
I'm on it.

ab1nash commented:
Now the buttons are visible only on selecting the message (and the usual hover as before) as requested.
Also I added support for night mode. 

![Screenshot from 2020-02-12 16-14-41](https://user-images.githubusercontent.com/41019632/74328136-64920f80-4db3-11ea-9aee-d82a50c43060.png)
![Screenshot from 2020-02-12 16-16-53](https://user-images.githubusercontent.com/41019632/74328138-65c33c80-4db3-11ea-8793-a823047d517e.png)

ab1nash commented:
@chrisbobbe @timabbott please review.

chrisbobbe commented on static/styles/night_mode.scss:
```diff
@@ -501,6 +501,12 @@ on a dark background, and don't change the dark labels dark either. */
     }
 
+    .selected_message .message_controls {
+        @media screen and (max-width: 350px) {
```
I don't see any other media queries in the night_mode file.

I also don't see `.night-mode` being used alongside `@media`, in any .scss file.

@showell, I wonder where might be the best place to put day and night styles that only apply on small screens; I'm not sure this is the best place.

timabbott replied:
Yeah, I'd prefer to just reference the night mode CSS class in the existing media CSS if we have to.

ab1nash commented:
Thanks for the insights @chrisbobbe !
I suppose I misunderstood the requirement and presumed that we only need to show those buttons when the message is selected (in mobile view). I will put the hover properties back.
Thank you for such meticulous testing. I will be more cautious with my testing before releasing a PR the next time.

ab1nash commented:
Yes the `line-height` is an unnecessary change. I did that to align the div container of the time element with that of the buttons. I will just remove it.

ab1nash commented:
@chrisbobbe `@media screen only` was causing that issue with hover. Fixed it. Made a few other changes.Moved the night mode css to zulip.css but I still think it is cleaner to have the night mode css separately as it seems redundant in there.

ab1nash commented:
Samples:
![Peek 2020-03-15 11-30](https://user-images.githubusercontent.com/41019632/76696362-8f240080-66b0-11ea-8131-8362cbea116a.gif)
![Peek 2020-03-15 11-31](https://user-images.githubusercontent.com/41019632/76696363-91865a80-66b0-11ea-8f9d-8f63d129113f.gif)

Hover opacity isn't changing in night mode. Working on it.

timabbott commented:
Closing as the bug in question was fixed in `main` by a different PR.
